streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableSource.java

```diff
@@ -86,12 +91,21 @@ public void init(final ProcessorContext context) {
             metrics = (StreamsMetricsImpl) context.metrics();
             droppedRecordsSensor = droppedRecordsSensorOrSkippedRecordsSensor(Thread.currentThread().getName(), context.taskId().toString(), metrics);
             if (queryableName != null) {
-                store = (TimestampedKeyValueStore<K, V>) context.getStateStore(queryableName);
+                final StateStore stateStore = context.getStateStore(queryableName);
+                try {
+                    store = ((WrappedStateStore<MeteredTimestampedKeyValueStore<K, V>, K, V>) stateStore).wrapped();
+                } catch (final ClassCastException e) {
+                    throw new IllegalStateException("Unexpected store type: " + stateStore.getClass() + " for store: " + queryableName, e);
+                }
                 tupleForwarder = new TimestampedTupleForwarder<>(
                     store,
                     context,
                     new TimestampedCacheFlushListener<>(context),
                     sendOldValues);
+                skippedIdempotentUpdatesSensor = skippedIdempotentUpdatesSensor(Thread.currentThread().getName(),
+                                                                                context.taskId().toString(),
+                                                                                ((InternalProcessorContext) context).currentNode().name(),
+                                                                                metrics);
```

Comment:
req: Please change formatting to comply with code style guidelines
```suggestion
                skippedIdempotentUpdatesSensor = skippedIdempotentUpdatesSensor(
                    Thread.currentThread().getName(), 
                    context.taskId().toString(), 
                    ((InternalProcessorContext) context).currentNode().name(), 
                    metrics
                );

```